llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -5225,7 +5225,10 @@ void SelectionDAGBuilder::visitTargetIntrinsic(const CallInst &I,
   // definition.
   const Function *F = I.getCalledFunction();
   bool HasChain = !F->doesNotAccessMemory();
-  bool OnlyLoad = HasChain && F->onlyReadsMemory();
+  bool OnlyLoad = HasChain && F->onlyReadsMemory() && F->willReturn();
+
+  bool PrevOnlyLoad = HasChain && F->onlyReadsMemory();
```

Comment:
Move this into the assert, unused variable. Or make onlyload in terms of PrevOnlyLoad (which isn't a great name either)

Reply:
`PrevOnlyLoad` and the assert are temporary, I wanted to use the test infrastructure to find cases where `OnlyLoad` would have been `true` and will now be `false` after the addition of the willReturn check.